Assets/Core/Weapon System/WeaponManager.cs

```diff
@@ -0,0 +1,133 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.Serialization;
+
+/// <summary>
+/// This class acts as the manager for equipped weapons, including input
+/// </summary>
+public class WeaponManager : MonoBehaviour
+{
+    
+    [Header("Pickup")]
+    public float pickupRange = 2.0f;
+    public LayerMask weaponLm;
+
+    public Transform weaponHolder;
+
+#if UNITY_EDITOR
+    [Header("Debug")]
+    [SerializeField] private bool log = false;
+#endif
+    
+    private bool _isWeaponHeld;
+    private bool _wantsToThrowOrGet;
+
+    private PlayerIA _playerInput;
+
+    //@TODO: Add 2 weapon support
+    private IWeapon _heldWeaponInterface;
+    private GameObject _heldWeaponGameObject;
+
+
+
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        _playerInput = new PlayerIA();
+        _playerInput.Gameplay.ThrowOrGet.performed += ThrowOrGetOnPerformed;
+        
+        _playerInput.Gameplay.ThrowOrGet.Enable();
```

Comment:
Remember `_playerInput.Gameplay.ThrowOrGet.Disable();` is needed on the OnDisabled function to avoid subscribing the action more than once or null reference errors